Index: Parser.php
===================================================================
--- Parser.php	(revision 56106)
+++ Parser.php	(working copy)
@@ -3304,6 +3308,8 @@
 				$output = "<$name$attrText/>";
 			} else {
 				$close = is_null( $params['close'] ) ? '' : $frame->expand( $params['close'] );
+				if ($this->needPreSave[$name])
+					return "<$name$attrText>".$this->replaceVariables($content).$close;
 				$output = "<$name$attrText>$content$close";
 			}
 		}
